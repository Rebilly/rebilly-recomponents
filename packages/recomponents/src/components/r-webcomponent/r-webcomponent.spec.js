import {mount} from '@vue/test-utils';
import {renderToString} from '@vue/server-test-utils';
import RWebcomponent from './r-webcomponent.vue';

describe('r-webcomponent.vue', () => {
    it('should render Wrapper and match snapshot', async () => {
        const wrapper = mount(RWebcomponent, {
            slots: {
                default: '<strong>default button</strong>',
            },
        });

        expect(wrapper.html()).toMatchSnapshot();
    });

    it('should render via SSR and match snapshot', async () => {
        const wrapper = renderToString(RWebcomponent, {
            slots: {
                default: '<strong>default button</strong>',
            },
        });

        expect(wrapper).toMatchSnapshot();
    });
});
