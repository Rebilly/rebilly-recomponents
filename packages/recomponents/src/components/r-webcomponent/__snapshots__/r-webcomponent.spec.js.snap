// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r-webcomponent.vue should render Wrapper and match snapshot 1`] = `<div class="r-webcomponent"><strong>default button</strong></div>`;

exports[`r-webcomponent.vue should render all specified components 1`] = `
<div class="r-webcomponent"><button role="button" title="" class="r-button r-button--size-regular r-button--type-default">
    <!----> Apply</button></div>
`;

exports[`r-webcomponent.vue should render via SSR and match snapshot 1`] = `<div class="r-webcomponent"><strong>default button</strong></div>`;
